protocol/value/kafka_string.go

```diff
@@ -0,0 +1,16 @@
+package value
+
+// KafkaString is a type of data structure whose encoding is done as
+// length -> 2 bytes, followed by string contents
+// It is different from compact string and compact nullable string
+type KafkaString struct {
```

Comment:
```suggestion
type String struct {
```

I think this is fine, wherever it is used it's going to be used as `value.String`, so wouldn't be confused for Go's `string`